### PR TITLE
Add configure-docker helper

### PR DESCRIPTION
**Pull request by doanac**
This allows you to do:

 sudo fioctl configure-docker

It will create a symlink from fioctl->/usr/bin/docker-credential-fio
and update your docker config with the helper. After that, docker
commands "just work".

Signed-off-by: Andy Doan <andy@foundries.io>

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -44,6 +45,11 @@ func Execute() {
 }
 
 func init() {
+	if os.Args[0] == docker.DOCKER_CREDS_HELPER {
+		initConfig()
+		os.Exit(docker.RunCredsHelper())
+	}
```
Comment (vkhoroz) [inline cmd/root.go:51]: It seems dangerous to me to exit from inside the `init` function (with success). In general, this behavior is undefined as there might be other `init` functions which will or will not be called in random order.
I'd recommend to move this logic into `func Execute`.

File: subcommands/docker/cmd.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+func NewCommand() *cobra.Command {
+	path, err := exec.LookPath("docker")
+	subcommands.DieNotNil(err, "Docker not found on system")
+	helperPath = filepath.Dir(path)
+
+	dockerConfigFile = dockerConfigPath()
+
+	cmd := &cobra.Command{
+		Use:   "configure-docker",
+		Short: "Configure a hub.foundries.io Docker credential helper",
```
Comment (vkhoroz) [inline subcommands/docker/cmd.go:34]: Some bigger help message about how to use the `creds-path` or run this in sudo mode would be helpful.

File: subcommands/docker/cmd.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+	dst := filepath.Join(helperPath, DOCKER_CREDS_HELPER)
+	fmt.Println("Symlinking", self, "to", dst)
+	subcommands.DieNotNil(os.Symlink(self, dst))
+
+	fmt.Println("Adding hub.foundries.io helper to", dockerConfigFile)
+	subcommands.DieNotNil(ioutil.WriteFile(dockerConfigFile, configBytes, 0o600))
+}
+
+func RunCredsHelper() int {
+	subcommands.Login(NewCommand())
```
Comment (vkhoroz) [inline subcommands/docker/cmd.go:104]: Please, add a comment how this call helps. IIUC it is to make sure that the access token is set properly, right?

But here is the problem. A user might also set the Token config parameter (I usually do that) in which case the AccessToken is empty. Need to workaround this somehow e.g. unset the Token attribute in Config.

File: subcommands/docker/cmd.go [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+	creds := struct {
+		Username string
+		Secret   string
+	}{
+		Username: "_fioctl_dch",
```
Comment (vkhoroz) [inline subcommands/docker/cmd.go:109]: This seem to work well; also please, consider this flow: https://raw.githubusercontent.com/google/go-containerregistry/main/images/credhelper-oauth.svg

**Comment by vkhoroz** [comment]
Looks good, just a few areas where I've spotted a place for improvement.

File: subcommands/docker/cmd.go [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+		Secret:   subcommands.Config.ClientCredentials.AccessToken,
+	}
+
+	bytes, err := json.Marshal(creds)
+	subcommands.DieNotNil(err)
+	os.Stdout.Write(bytes)
+	return 0
```
Comment (vkhoroz) [inline subcommands/docker/cmd.go:116]: The https://docs.docker.com/engine/reference/commandline/login/#credential-helpers documentation is very sparse, but what I understood from it as well as the google diagrams (see my other comments) - docker login will send a `docker-credential-fio get` command when it needs to login. Maybe, it makes sense to check for the `get` subcommand here explicitly; and bail out if docker sends some other (unexpected) subcommand.

File: cmd/root.go (hunk 1 above)
Comment (doanac) [reply cmd/root.go]: fixed in force-push

File: subcommands/docker/cmd.go (hunk 2 above)
Comment (doanac) [reply subcommands/docker/cmd.go:34]: added more in the force-push - see what you think

File: subcommands/docker/cmd.go (hunk 3 above)
Comment (doanac) [reply subcommands/docker/cmd.go]: force-push includes a comment. it also has a check - it will exit with an error if oauth2 creds aren't being used

**Comment by vkhoroz** [comment]
LGTM

**Pull request merged**